### PR TITLE
Update go.mod to remove vulnerabilities

### PR DESCRIPTION
**Pull request by spotlesstofu**
Apply the updates suggested by `govulncheck`:
- go 1.22.7
- github.com/coreos/ignition/v2@v2.14.0

Also run `go mod tidy`.

### CODE REVIEW

**Comment by spotlesstofu** [comment]
I see tests are failing with this:
```
go: go.mod requires go >= 1.22.7 (running go 1.21.13; GOTOOLCHAIN=local)
```

**Comment by littlejawa** [comment]
We are referencing a golang builder image in Dockerfile and Makefile that is still 1.21
You probably need to updated it to `rhel-9-golang-1.22-openshift-4.17` instead

File: go.mod [hunk 1]
```diff
@@ -1,18 +1,19 @@
 module github.com/openshift/sandboxed-containers-operator
 
-go 1.21
+go 1.22.7
```
Comment (beraldoleal) [inline go.mod:3]: This is the reason of the build is failing... The CI environment should update go toolchain or set GOTOOLCHAIN=auto, that will download automatically to satisfy this minimal requirement.

**Review by beraldoleal — COMMENTED** [review]
Hi @spotlesstofu , this lgtm, just the CI is failing to build because know we are using 1.22.7.

**Comment by beraldoleal** [comment]
We also need to backport this to release-1.8 after merge.

File: go.mod (hunk 1 above)
Comment (gkurz) [reply go.mod:3]: I prefer to bump the go toolchain manually for now and keep CI improvements for the next cycle.

https://github.com/openshift/release/pull/59339

**Comment by gkurz** [comment]
> We also need to backport this to release-1.8 after merge.

We'll have several items to backport. I'll regroup that in a single PR when everything is merged into devel.

**Review by gkurz — APPROVED** [review]
/lgtm

`govulncheck` is happy after this change. Thanks @spotlesstofu !

We'll rerun CI when https://github.com/openshift/release/pull/59339 is merged.

**Comment by gkurz** [comment]
https://github.com/openshift/release/pull/59339 got merged at last !

/retest

We're now expecting `ci/prow/check` to pass and `ci/prow/sandboxed-containers-operator-e2e` to fail (see https://github.com/openshift/release/pull/59339#issuecomment-2507814251 for details).

**Review by beraldoleal — APPROVED** [review]
LGTM, thanks

**Comment by gkurz** [comment]
Fixes: https://issues.redhat.com/browse/OCPBUGS-36878
Fixes: https://issues.redhat.com/browse/OCPBUGS-41472
Fixes: https://issues.redhat.com/browse/OCPBUGS-41410

**Comment by gkurz** [comment]
> We're now expecting `ci/prow/check` to pass and `ci/prow/sandboxed-containers-operator-e2e` to fail (see https://github.com/openshift/release/pull/59339#issuecomment-2507814251 for details).

Outcome is as expected. Let's merge !

**Pull request merged**